examples/prodlda.py

```diff
@@ -0,0 +1,369 @@
+
+    @nn.compact
+    def __call__(self, inputs, is_training):
+        h = nn.softplus(nn.Dense(self.hidden)(inputs))
+        h = nn.softplus(nn.Dense(self.hidden)(h))
+        h = nn.Dropout(self.dropout_rate, deterministic=not is_training)(h)
+
+        h_mu, h_sigma = nn.Dense(self.num_topics)(h), nn.Dense(self.num_topics)(h)
+
+        logtheta_loc = nn.BatchNorm(
+            use_bias=False,
+            use_scale=False,
+            momentum=0.9,
+            use_running_average=not is_training,
+        )(h_mu)
+        logtheta_logvar = nn.BatchNorm(
+            use_bias=False,
+            use_scale=False,
+            momentum=0.9,
+            use_running_average=not is_training,
+        )(h_sigma)
+        logtheta_scale = jnp.exp(0.5 * logtheta_logvar)
+        return logtheta_loc, logtheta_scale
+
+
+class FlaxDecoder(nn.Module):
+    vocab_size: int
+    dropout_rate: float
+
+    @nn.compact
+    def __call__(self, inputs, is_training):
+        h = nn.Dropout(self.dropout_rate, deterministic=not is_training)(inputs)
+        h = nn.Dense(self.vocab_size, use_bias=False)(h)
+        h = nn.BatchNorm(
+            use_bias=False,
+            use_scale=False,
+            momentum=0.9,
+            use_running_average=not is_training,
+        )(h)
+        return nn.softmax(h)
+
+
+def model(docs, hyperparams, is_training=False, nn_framework="flax"):
+    if nn_framework == "flax":
+        decoder = flax_module(
+            "decoder",
+            FlaxDecoder(hyperparams["vocab_size"], hyperparams["dropout_rate"]),
+            input_shape=(1, hyperparams["num_topics"]),
+            # ensure PRNGKey is made available to dropout layers
+            apply_rng=["dropout"],
+            # indicate mutable state due to BatchNorm layers
+            mutable=["batch_stats"],
+            # to ensure proper initialisation of BatchNorm we must
+            # initialise with is_training=True
+            is_training=True,
+        )
+    elif nn_framework == "haiku":
+        decoder = haiku_module(
+            "decoder",
+            # use `transform_with_state` for BatchNorm
+            hk.transform_with_state(
+                HaikuDecoder(hyperparams["vocab_size"], hyperparams["dropout_rate"])
+            ),
+            input_shape=(1, hyperparams["num_topics"]),
+            apply_rng=True,
+            # to ensure proper initialisation of BatchNorm we must
+            # initialise with is_training=True
+            is_training=True,
+        )
+    else:
+        raise ValueError(f"Invalid choice {nn_framework} for argument nn_framework")
+
+    with numpyro.plate(
+        "documents", docs.shape[0], subsample_size=hyperparams["batch_size"]
+    ):
+        batch_docs = numpyro.subsample(docs, event_dim=1)
+        logtheta = numpyro.sample(
+            "logtheta",
+            dist.Normal(0, 1).expand((hyperparams["num_topics"],)).to_event(1),
+        )
+        theta = jax.nn.softmax(logtheta)
```

Comment:
This can be changed into
```diff
-        logtheta = numpyro.sample(
-            "logtheta",
-            dist.Normal(0, 1).expand((hyperparams["num_topics"],)).to_event(1),
+        theta = numpyro.sample(
+            "theta",
+            dist.Dirichlet(jnp.ones(hyperparams["num_topics"]))
,
         )
-        theta = jax.nn.softmax(logtheta)
```